lib/permissions.js

```diff
@@ -597,38 +484,26 @@ exports = module.exports = {
    */
 
   policyChecker: function () {
-    var that = this;
-    return function (req, res, next) {
-      // first check if the route should be ignored
-      if (that.shouldIgnore(req.url)) {
-        next();
-        return;
+    return (req, res, next) => {
+      if(this.shouldIgnore(req.url)) {
+        return next(); // route has been explicitly ignored somewhere
       }
+      const user = app.usermanager.getCurrentUser();
 
-      var user = app.usermanager.getCurrentUser();
       if (!user || !user._id) {
-        res.statusCode = 403;
-        return res.json({statusCode: STATUS.NOT_AUTHENTICATED});
+        return res.status(403).json({ statusCode: STATUS.NOT_AUTHENTICATED });
       }
+      const action = this.getActionType(req.method.toLowerCase());
+      const resource = this.buildResourceString(user.tenant._id, req.path);
 
-      // build path from req.url
-      var resource = that.buildResourceString(user.tenant._id, req.url);
-      var action = that.getActionType(req.method.toLowerCase()); // at this point, we can only enforce CRUD methods
-      that.hasPermission(user._id, action, resource, function (error, allowed) {
-        if (error) {
-          // summat went wrong
-          res.statusCode = 500;
-          return res.json(error);
+      this.hasPermission(user._id, action, resource, function (error, allowed) {
+        if(error) {
+          return res.status(500).json(error);
         }
-
-        if (allowed) {
-          // if we have permission, just continue
-          return next();
+        if(!allowed) {
+          res.status(403).json({ statusCode: STATUS.ACCESS_DENIED });
```

Comment:
Should this return here?

I'm getting the error `Error: Can't set headers after they are sent.`pointing to this line if I try and preview as a course creator.

Reply:
```suggestion
          return res.status(403).json({ statusCode: STATUS.ACCESS_DENIED });
```

Reply:
Good spot, have added

Reply:
Cool, is it parsing the statements for actions correctly [here](https://github.com/adaptlearning/adapt_authoring/blob/issue/2028/lib/permissions.js#L139)? When I login as a course creator and try and preview my own course it cannot find read action in any statements and an error is returned.